sparse/mlir_backend/_common.py

```diff
@@ -12,6 +12,10 @@ class MlirType(abc.ABC):
     def get_mlir_type(cls) -> ir.Type: ...
 
 
+class RefableList(list):
```

Comment:
Rename to `PackedArgumentTuple(tuple)`?

Reply:
Unfortunately we can't subclass `tuple` here. This class is used in `weakref.finalize(...)` and the `weakref` docs say:

> Several built-in types such as [list](https://docs.python.org/3/library/stdtypes.html#list) and [dict](https://docs.python.org/3/library/stdtypes.html#dict) do not directly support weak references but can add support through subclassing:
> ```python
> class Dict(dict): 
>     pass
> obj = Dict(red=1, green=2, blue=3)   # this object is weak referenceable
> ```
> CPython implementation detail: Other built-in types such as [tuple](https://docs.python.org/3/library/stdtypes.html#tuple) and [int](https://docs.python.org/3/library/functions.html#int) do not support weak references even when subclassed.

So it must be a `list` subclass. Would you still like to change its name?

Reply:
That sounds okay, but let's `s/List/Tuple` instead.

Reply:
What do you mean by `s/Tuple/List`? Subclass `List` from `typing`?

Reply:
I was initially thinking of renaming, but how about composition (using `tuple`) instead of inheritance? It will solve the `weakref` issue; and be a bit more future proof. If needed, we can define `__getitem__`, `__len__` and `__iter__` for ease of usage. And let's use a `dataclass`.

Reply:
Sure! I moved to a `@dataclass` here, and implemented functions required for the iteration. I also renamed it to `PackedArgumentTuple`.